internal/lookoutv2/repository/getjobrunerror.go

```diff
@@ -29,24 +29,16 @@ func NewSqlGetJobRunErrorRepository(db *pgxpool.Pool, decompressor compress.Deco
 
 func (r *SqlGetJobRunErrorRepository) GetJobRunError(ctx context.Context, runId string) (string, error) {
 	var rawBytes []byte
-	err := r.db.BeginTxFunc(ctx, pgx.TxOptions{
+	err := pgx.BeginTxFunc(ctx, r.db, pgx.TxOptions{
 		IsoLevel:       pgx.RepeatableRead,
 		AccessMode:     pgx.ReadOnly,
 		DeferrableMode: pgx.Deferrable,
 	}, func(tx pgx.Tx) error {
-		rows, err := tx.Query(ctx, "SELECT error FROM job_run WHERE run_id = $1 AND error IS NOT NULL", runId)
-		if err != nil {
-			return err
+		err := tx.QueryRow(ctx, "SELECT error FROM job_run WHERE run_id = $1 AND error IS NOT NULL", runId).Scan(&rawBytes)
+		if err == pgx.ErrNoRows {
+			return errors.Errorf("no error found for run with id %s", runId)
 		}
-		defer rows.Close()
-		for rows.Next() {
-			err := rows.Scan(&rawBytes)
-			if err != nil {
-				return err
-			}
-			return nil
-		}
-		return errors.Errorf("no error found for run with id %s", runId)
+		return err
 	})
```

Comment:
I noticed that the code here, and other parts, used `tx.Query` just to query a single row and found it confusing. switched to `tx.QueryRow` instead.